### PR TITLE
prepare script

### PR DESCRIPTION
**Pull request by costa-e**

### CODE REVIEW

File: evaluation/evaluation_readme.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+
+
+# Evaluation for CTA-oneAPI
+
+This folder contains the scripting to evaluate the tool.
+Specifically, 2 things are done for evaluation:
+* evaluate impact on all sample projects of oneAPI-directProgramming repo ()
+* evaluate impact on real world repo relevant for the health sector, Relion ()
```
Comment (Wenqi997) [inline evaluation/evaluation_readme.md:8]: Dose "( )" after repo and Relion prepare to contain the repo website?

File: auto_editor/Fix1003Rule.py [hunk 2]
```diff
@@ -22,78 +22,87 @@ def change_type(self) -> ChangeTypeEnum:
...
                 one_line_warning_code = False
             else:
-                now_code = self.strip_determin( first_time, self.all_lines[i])
+                now_code = self.strip_determin(first_time, self.all_lines[i])
                 if one_line_warning_code == False:
                     warning_code = warning_code + now_code + "\n"
                 else:
                     warning_code = now_code
-                print("code to remove: ",warning_code)
+                print("code to remove: ", warning_code)
```
Comment (Wenqi997) [inline auto_editor/Fix1003Rule.py:48]: can remove this one

File: auto_editor/Fix1003Rule.py [hunk 3]
```diff
@@ -22,78 +22,87 @@ def change_type(self) -> ChangeTypeEnum:
...
+
+                # remove old statement
+                self.remove_obsolete_code(i, warning_last_line, self.all_lines, other_warnings)
 
-                warning_begin_index = self.replace_useless_multiple_line(warning_begin_index-1, i, self.all_lines)
+                # remove old warning message
+                self.remove_code(self.all_lines, warning_first_line, warning_last_line)
 
-                print('warning_begin_index',warning_begin_index)
+                print('warning_begin_index', warning_first_line)
```
Comment (Wenqi997) [inline auto_editor/Fix1003Rule.py:59]: can remove

File: auto_editor/Fix1003Rule.py [hunk 4]
```diff
@@ -22,78 +22,87 @@ def change_type(self) -> ChangeTypeEnum:
...
                 new_lineItem = LineItem(new_code)
-                self.all_lines.insert(warning_begin_index,new_lineItem)
+                self.all_lines.insert(warning_first_line, new_lineItem)
                 tmp_dict[file_path] = self.all_lines
                 break
 
         project.paths_to_lines[file_path] = tmp_dict[file_path]
-        print('length: ',len(project.paths_to_lines[file_path]))
+        print('length: ', len(project.paths_to_lines[file_path]))
```
Comment (Wenqi997) [inline auto_editor/Fix1003Rule.py:66]: remove

File: auto_editor/Fix1003Rule.py [hunk 5]
```diff
@@ -22,78 +22,87 @@ def change_type(self) -> ChangeTypeEnum:
...
         warning_code, prefix = "", ""
         tmp_dict = project.paths_to_lines
         self.all_lines = tmp_dict[file_path]
-        warning_begin_id = self.all_lines[warning_first_line].id
-        warning_end_id = self.all_lines[warning_last_line].id
-        warning_begin_index = get_index_of_line_id(warning_begin_id, self.all_lines)
-        warning_end_index = get_index_of_line_id(warning_end_id, self.all_lines)
-        print('warning_first_line: ',warning_first_line,' ; warning_last_line: ',warning_last_line)
+        other_warnings = self.get_other_warnings(warning_first_line, project)
+        print('warning_first_line: ', warning_first_line, ' ; warning_last_line: ', warning_last_line)
```
Comment (Wenqi997) [inline auto_editor/Fix1003Rule.py:31]: can remove

File: auto_editor/BaseRule.py [hunk 6]
```diff
@@ -37,6 +37,8 @@ def initiate_run(self, project: StructuredProjectSource):
 
                 # index of line id is taken at each iteration,
                 # as it may change if the project is modified in previous iterations
+                if file_path == 'shortcuts.dp.hpp':
+                    print('yo')
```
Comment (Wenqi997) [inline auto_editor/BaseRule.py:41]: why print ‘yo’ there？

**Review by Wenqi997 — COMMENTED** [review]
You really thought this through. That's amazing！！

File: auto_editor/BaseRule.py (hunk 6 above)
Comment (costa-e) [reply auto_editor/BaseRule.py:41]: haha oh, that was left over from debugging

File: evaluation/evaluation_readme.md (hunk 1 above)
Comment (costa-e) [reply evaluation/evaluation_readme.md:8]: oh yes, added it. Thx!

File: auto_editor/Fix1003Rule.py [hunk 7]
```diff
@@ -28,11 +28,9 @@ def run_rule(self, project: StructuredProjectSource,
         tmp_dict = project.paths_to_lines
```
Comment (Wenqi997) [inline auto_editor/Fix1003Rule.py:28]: remove “print（“----------------------------”）” or left some print info below

**Review by Wenqi997 — COMMENTED** [review]
do you think we need to leave some print info? So that we can see how CTA going on.  Nothing shows also good.

File: auto_editor/Fix1003Rule.py (hunk 7 above)
Comment (costa-e) [reply auto_editor/Fix1003Rule.py:27]: good point, some output is probably helpful because it can take some time. I added a message

**Review by Wenqi997 — COMMENTED** [review]
great!!!

**Pull request merged**